src/pages/SwapV2/index.tsx

```diff
@@ -629,17 +629,19 @@ export default function Swap({ history }: RouteComponentProps) {
     if (isSelectCurencyMannual) syncUrl(currencyIn, currencyOut) // when we select token manual
   }, [currencyIn, currencyOut, isSelectCurencyMannual, syncUrl])
 
+  const prevChainId = usePrevious(chainId) || chainId
+  const justChangeNetwork = prevChainId !== chainId
```

Comment:
This code made `justChangeNetwork` flip-flop between `true` and `false`, which `true` makes useEffect run, but `false` not and make useless re-render. I suggest make it a variable which count times of chainId changed `chainIdChangedCount` and useEffect gonna watch on that variable to run effect
```
useEffect(() => {
  ...
  // disable react exhaustive deps
}, [chainIdChangedCount])
```

Reply:
I guess it only changes from `false` to `true` when switching networks 🤔

Reply:
> I guess it only changes from `false` to `true` when switching networks 🤔

yes. correct

Reply:
then after that, one re-render happened which is not happened by change network, usePrevious return same chainId with current one, which make it change again from `true` to false`